frontend/src/Initialize.tsx

```diff
@@ -216,6 +217,146 @@ const Initialize = (props: Props) => {
+      let kittySizeSet = false;
+      let kittySize = null;
+      for (const [key, value] of Object.entries(gameSettings)) {
+        switch (key) {
+          case "game_mode":
+            send({
+              Action: {
+                SetGameMode: value,
+              },
+            });
+            break;
+          case "num_decks":
+            send({
+              Action: {
+                SetNumDecks: value,
+              },
+            });
+            if (kittySizeSet) {
+              // reset the size again, as setting deck numn resets kitty_size to default
+              send({
+                Action: {
+                  SetKittySize: kittySize,
+                },
+              });
+            }
+            break;
+          case "kitty_size":
+            send({
+              Action: {
+                SetKittySize: value,
+              },
+            });
+            kittySizeSet = true;
+            kittySize = value;
+            break;
+          case "friend_selection_policy":
+            send({
+              Action: {
+                SetFriendSelectionPolicy: value,
+              },
+            });
+            break;
+          case "hide_landlord_points":
+            send({
+              Action: {
+                SetHideLandlordsPoints: value,
+              },
+            });
+            break;
+          case "hide_played_cards":
+            send({ Action: { SetHidePlayedCards: value } });
+            break;
+          case "advancement_policy":
+            send({
+              Action: {
+                SetAdvancementPolicy: value,
+              },
+            });
+            break;
+          case "kitty_bid_policy":
+            send({
+              Action: {
+                SetKittyBidPolicy: value,
+              },
+            });
+            break;
+          case "kitty_penalty":
+            send({
+              Action: {
+                SetKittyPenalty: value,
+              },
+            });
+            break;
+          case "throw_penalty":
+            send({
+              Action: {
+                SetThrowPenalty: value,
+              },
+            });
+            break;
+          case "trick_draw_policy":
+            send({
+              Action: {
+                SetTrickDrawPolicy: value,
+              },
+            });
+            break;
+          case "throw_evaluation_policy":
+            send({
+              Action: {
+                SetThrowEvaluationPolicy: value,
+              },
+            });
+            break;
+        }
+      }
+    }
+  };
+
+  const loadGameSettings = (evt: any) => {
+    evt.preventDefault();
+    const settings = localStorage.getItem("gameSettingsInLocalStorage");
+    if (settings !== null) {
+      let gameSettings: IPropagatedState;
+      try {
+        gameSettings = JSON.parse(settings);
+        setGameSettings(gameSettings);
+      } catch (err) {
+        localStorage.setItem(
```

Comment:
in the error case, we probably want to unset the value in localstorage (rather than setting it to the current value)?